apps/admin-gui/src/app/shared/components/dialogs/copy-sponsored-members-dialog/copy-sponsored-members-dialog.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+  MembersManagerService,
+  MemberWithSponsors,
+  RichUser,
+  User,
+} from '@perun-web-apps/perun/openapi';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import {
+  NotificatorService,
+  PerunTranslateService,
+  StoreService,
+} from '@perun-web-apps/perun/services';
+import { SelectionModel } from '@angular/cdk/collections';
+import { TABLE_ADD_SPONSORED_MEMBERS } from '@perun-web-apps/config/table-config';
+import { formatDate } from '@angular/common';
+
+export interface CopySponsoredMembersDialogData {
+  voId: number;
+  theme: string;
+  voSponsors: RichUser[];
+  findSponsorsAuth: boolean;
+}
+@Component({
+  selector: 'app-copy-sponsored-members-dialog',
+  templateUrl: './copy-sponsored-members-dialog.component.html',
+  styleUrls: ['./copy-sponsored-members-dialog.component.scss'],
+})
+export class CopySponsoredMembersDialogComponent implements OnInit {
+  loading = false;
+  theme: string;
+  sponsorType: string;
+  tableId = TABLE_ADD_SPONSORED_MEMBERS;
+  voSponsors: RichUser[];
+  sponsorFrom: User;
+  sponsorTo: User;
+  sponsoredMembers: MemberWithSponsors[];
+  filteredSponsoredMembers: MemberWithSponsors[];
+  selection: SelectionModel<MemberWithSponsors> = new SelectionModel<MemberWithSponsors>(true, []);
+  firstSearchDone = false;
+  filter: string;
+  expiration = 'never';
+  pickValidity = false;
+  disableSelf = false;
+  constructor(
+    private dialogRef: MatDialogRef<CopySponsoredMembersDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: CopySponsoredMembersDialogData,
+    private store: StoreService,
+    private membersService: MembersManagerService,
+    private notificator: NotificatorService,
+    private translate: PerunTranslateService
+  ) {}
+
+  ngOnInit(): void {
+    this.membersService.getAllSponsoredMembersAndTheirSponsors(this.data.voId, []).subscribe({
+      next: (members) => {
+        this.sponsoredMembers = members;
+      },
+    });
+    this.voSponsors = this.data.voSponsors;
+    this.theme = this.data.theme;
+  }
+
+  sponsorFromChanged(user: User): void {
```

Comment:
I like the terms of "source" and "target" sponsor which you used in the dialog itself. I think it would make sense to use these terms in the code as well.